ext/cache/lib.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+mod sqlite;
+pub use sqlite::SqliteBackedCache;
+
+use async_trait::async_trait;
+use deno_core::error::AnyError;
+use deno_core::include_js_files;
+use deno_core::op;
+use deno_core::serde::Deserialize;
+use deno_core::serde::Serialize;
+use deno_core::Extension;
+use deno_core::OpState;
+use deno_core::Resource;
+use deno_core::ResourceId;
+
+use std::cell::RefCell;
+use std::path::PathBuf;
+use std::rc::Rc;
+use std::sync::Arc;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct CachePutRequest {
+  pub cache_id: i64,
+  pub request_url: String,
+  // TODO(@satyarohith): use ByteString for headers.
+  // Note: ByteString cannot be stored as JSON in sqlite.
+  // Serialize into a buffer with V8 serializer.
```

Comment:
Just FYI serialization format is not stable across V8 releases

Reply:
The format is not stable, but it is backwards compatible.